ci/script.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Echo all commands before executing them
+set -o xtrace
+# Forbid any unset variables
+set -o nounset
+# Exit on any error
+set -o errexit
+
+# Ensure there are no outstanding lints.
+check_lints() {
+    ## In the future, it would be good if `|| true` can be removed so that
+    ## clippy warnings abort the build.
+    cargo clippy --all --features "$FEATURES" || true
+}
+
+# Ensure the code is correctly formatted.
+check_format() {
+    cargo fmt --all -- --check
+}
+
+# Run the test suite.
+check_tests() {
+    cargo test --all --examples --tests --benches --no-default-features
+    cargo test --all --examples --tests --benches --features "$FEATURES"
```

Comment:
IIUC, the commands above don't run all the tests. (They miss the ones in `serialization-tests`, `blas-tests`, and `numeric-tests`.)

As I see it, we have two options:

1. Keep the existing `serialization-tests`, `blas-tests`, and `numeric-tests` crates and run their tests in the scripts.

2. Merge them into the `ndarray` crate, adding the necessary optional dependencies and creating the necessary test-specific features in `ndarray`, similar to how I'm suggesting to handle the `test-blas-openblas-src` feature.

I think option 2 is a bit cleaner, but either is fine with me. Either way, we need to make sure that we run these tests.

Reply:
I created an issue about specifically this in #637.  I agree that keeping these scripts separate isn't nice and integrating them I think would be the best option.